xarray/backends/zarr.py

```diff
@@ -679,8 +686,24 @@ def set_variables(self, variables, check_encoding_set, writer, unlimited_dims=No
 
                 if coding.strings.check_vlen_dtype(dtype) == str:
                     dtype = str
+
+                if self._write_empty is not None:
+                    if (
+                        "write_empty_chunks" in encoding
+                        and encoding["write_empty_chunks"] != self._write_empty
+                    ):
+                        raise ValueError(
+                            'Differing "write_empty_chunks" values in encoding and parameters'
```

Comment:
```suggestion
                            'Differing "write_empty_chunks" values in encoding and parameters'
                            f'Got {encoding["write_empty_chunks"] = } and {self._write_empty = }'
```
Make the error a little more helpful by showing the value of each?